builds/e2e/templates/e2e-run.yaml

```diff
@@ -77,8 +77,15 @@ steps:
     {
       $filter += '&Category!=NestedEdgeOnly'
     }
-
-    sudo --preserve-env dotnet test $testFile --no-build --logger 'trx' --filter "$filter"
+    
+    if ('$(artifactName)'.Contains('centos'))
+    {
+      sudo --preserve-env $(command -v dotnet) test $testFile --no-build --logger 'trx' --filter "$filter"
+    }
+    else
+    {
+      sudo --preserve-env dotnet test $testFile --no-build --logger 'trx' --filter "$filter"
+    }
```

Comment:
I think you should just always use the full non-sudo path to invoke the tests for all platforms, rather than conditionalizing on "centos".

Reply:
I tried that , but I ran into a failure 

https://dev.azure.com/msazure/One/_build/results?buildId=48812079&view=logs&j=9ec97737-d75d-57d0-984a-aca7d53a27f6&t=ad31136c-cded-5a78-04ec-3842c2d45022